### PR TITLE
fix: handle NotFoundError(#218)

### PR DESCRIPTION
**Pull request by BlackGlory**
https://github.com/fastify/fastify-static/issues/218

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Eomm** [comment]
Could you add a test to avoid regressions?
Thank you

**Comment by BlackGlory** [comment]
The testing of this project is too confusing for me, I can't do anything about it.
I hope someone can find a way to correctly locate this error from the example I gave, and then write a proper test.

File: index.js [hunk 1]
```diff
@@ -187,6 +187,12 @@ async function fastifyStatic (fastify, opts) {
 
         return reply.callNotFound()
       }
+
+      // err is an object created by `http-errors`
+      if (err.status === 404) {
```
Comment (climba03003) [inline index.js:192]: If the status is 404 and it is created by `http-errors`. Then it should be good to go with the below line `reply.send(err)`. Isn't it?

Comment (climba03003) [reply index.js:192]: I see where is the problem. ignore this comment.

**Comment by Eomm** [comment]
Superseded by #225

**Pull request closed without merging**